### DISCUSSION TITLE
Improve ability to discuss - Add Model Comments, Review Reactions, and Review Comments

### DISCUSSION THREAD

**Discussion by JustMaier** [civitai/civitai#20] [Ideas]
## Model Comments
This would be a good place for people to share their prompts, tips regarding specific model etc.

I imagine it could either be added in one of the following ways:
- Under reviews (we'd probably have to page reviews or make them scroll horizontally)
- As a vertical tab along-side reviews
- Intermixed with reviews as a revise "Community Feedback" area.

Functionally I'd like to see:
- Threaded comments
- Upvoting or Reacting
- Rich Text Editor

## Review Reactions
People need a way to recognize each other's reviews. Let's add a button for adding a reaction and then show the number of those reactions on reviews. Once a reaction is showing on a review, another user can click the reaction to add/remove their reaction. If a user isn't logged in, we should show a popover similar to what we are doing when an unauthenticated user tries to show an NSFW image.

Let's start with the following reactions:
- Like
- Dislike
- Laugh
- Cry
- Heart

**This is live!**
![image](https://user-images.githubusercontent.com/607609/202655182-b2b0ed55-37d0-46f0-85ed-a7722d6fdca4.png)

## Review Comments
Sometimes people might leave questions in their reviews, or creators might want to comment on a review or give feedback. I imagine these wouldn't be threaded just to keep things simple.

**Comment by JustMaier** [comment]
@manuelurenah Is going to be taking the Review Reactions portion of this.

## Super rough mock-up of Review Reactions
![image](https://user-images.githubusercontent.com/607609/201741599-4c2df3cd-4485-4c3f-b263-da9fdec1a22c.png)

- [x] At the bottom of a review card add a "Add Reaction" button. On click display a popover with the reactions they can add.
- [x] At the bottom of a review card display current reaction counts. On click toggle the reaction for the current user. If the current user has reacted, show a blue outline (or something)
- [ ] Add the ability to sort by each reaction (low-pri stretch)

**Reply by manuelurenah** [reply]
Happy to announce this feature has been implemented and is working nicely, you can see the changes here https://github.com/civitai/civitai/commit/588d9342d3cad22e71177a555a741afe4491a54e and here https://github.com/civitai/civitai/commit/79c5143750adbcde344212f82a7158c9683f9c3e 🙂 If there are any issues with it or improvements that need to be made, happy to work on this again

**Comment by JustMaier** [comment]
@manuelurenah Here's the idea for the updated details page that includes comments on reviews and models:
https://github.com/civitai/civitai/discussions/26

**Comment by manuelurenah** [comment]
We now have a PR with the changes required to supporting commenting in models and reviews, check it out https://github.com/civitai/civitai/pull/71

**Comment by manuelurenah** [comment]
The commenting system is now live! 🚀 Users are should be able to comment on models and reply to these comments and reviews as well. Here are a couple of screenshots of how it looks like

![SCR-20221207-ipf](https://user-images.githubusercontent.com/12631159/206252393-30f245ff-f52f-4707-855e-bf21d73036c1.jpeg)
![SCR-20221207-irb](https://user-images.githubusercontent.com/12631159/206252407-34c43c63-ab10-4b98-94c3-9400244f0ddb.jpeg)
![SCR-20221207-irr](https://user-images.githubusercontent.com/12631159/206252414-d9e6fd38-a5f0-4da9-91e7-80a320544036.jpeg)
![SCR-20221207-is8](https://user-images.githubusercontent.com/12631159/206252418-f1a6c69d-3b63-419e-915a-658b4129ef7c.jpeg)
